src/components/ebay-table/component.ts

```diff
@@ -0,0 +1,26 @@
+import { WithNormalizedProps } from "../../global";
+
+// export type ColumnType = "normal" | "numeric" | "row-header" | "none";
+export type TableDensity = "compact" | "relaxed" | "none";
+export interface TableColumn {
+    type?: string; // Use ColumnType after marko fix the ts error for attr tags
+    header: Marko.AttrTag<TableHeader> | string;
+}
+export interface TableHeader extends Omit<Marko.Input<"th">, `on${string}`> {
+    renderBody: Marko.Body;
+}
+export interface TableCell
+    extends Omit<Marko.Input<"td"> & Marko.Input<"th">, `on${string}`> {
+    renderBody: Marko.Body;
```

Comment:
Mentioned in my other comment— would it make any sense to add `type: "layout" | "multiline" | "thumbnail"` here and remove the helpers in `components/`?

Reply:
I tried, but just for these 3, we have many kind of combinations, say `mutliline` can be with or without `layout`, and thumbnail can be first element or second element of layout or without layout. We are going to have more cell components and I think eventually we will not be able to deliver everything through attr tags. @agliga what do you think?

Reply:
Discussed with @agliga and will do these in separate PR.